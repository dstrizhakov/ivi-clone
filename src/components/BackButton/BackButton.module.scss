@import '@/styles/variables.scss';
@import '@/styles/mixins.scss';
.back {
  cursor: pointer;
  background-color: transparent;
  color: white;
  display: flex;
  align-items: center;
  opacity: 0.7;
  transition: $time-main;
  margin: rem(15) 0;
  &__icon {
    margin-left: rem(-10);
    width: rem(35);
    height: rem(35);
  }
  & span {
    font-size: rem(15);
  }
  &:hover {
    opacity: 1;
    margin: rem(15) 0;
  }
  &:hover .back__icon {
    transform: scale(1.1);
  }
}
