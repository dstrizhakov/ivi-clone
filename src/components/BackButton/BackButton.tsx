import { useRouter } from 'next/navigation';
import { HiChevronLeft } from 'react-icons/hi';
import styles from './BackButton.module.scss';
import { useTranslation } from 'react-i18next';
import { useEscapeKey } from '@/hooks/useEscapeKey';
import { FC } from 'react';

const BackButton: FC = (): JSX.Element => {
  const { t } = useTranslation();
  const router = useRouter();

  const back = () => {
    router.back();
  };

  useEscapeKey(back);

  return (
    <button className={styles.back} onClick={back}>
      <HiChevronLeft className={styles.back__icon} />
      <span>{t('buttons.back')}</span>
    </button>
  );
};

export default BackButton;
