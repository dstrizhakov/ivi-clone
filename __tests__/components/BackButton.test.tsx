import { render } from '@testing-library/react';
import React from 'react';
import BackButton from '../../src/components/BackButton/BackButton';

describe('BackButton', () => {
  it('should be as in snapshot', () => {
    const component = render(<BackButton />);
    expect(component).toMatchSnapshot();
  });
});
