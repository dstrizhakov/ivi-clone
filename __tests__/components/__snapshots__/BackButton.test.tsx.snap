// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton should be as in snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="back"
      >
        <svg
          aria-hidden="true"
          class="back__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"
            fill-rule="evenodd"
          />
        </svg>
        <span>
          buttons.back
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="back"
    >
      <svg
        aria-hidden="true"
        class="back__icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"
          fill-rule="evenodd"
        />
      </svg>
      <span>
        buttons.back
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
